provisioning.adoc

```diff
@@ -0,0 +1,303 @@
+// vim: tw=79
+= Provisioning with Tendrl
+
+:toc:
+
+== Layers of Provisioning
+
+. Bare-metal provisioning brings up the OS on the storage nodes in the specific
+network(s).
+. Tendrl provisioning sets up tendrl api, etcd and node agents on the
+administrator specified storage nodes. It is possible that this could be
+accomplished irrespective of tendrl’s provisioning modules, either by the bare
+metal system directly or as part of image based or template based deployments.
+
+At this point, tendrl itself is up and running and will have imported the host
+information such as cpu, ram, block storage information etc.
+
+Once tendrl is up and running, administrator either imports or creates a new
+cluster. The rest of the document discusses how these operations would be
+implemented via tendrl.
+
+
+== Components
+
+The following components are referred to in the rest of this document.
+
+Provisioner::
+This is the ansible based provisioning system that tendrl relies upon to
+install and upgrade the storage system. Tendrl takes care of setting up the
+necessary user accounts and ssh access to enable this system to function. This
+system is invoked by tendrl as part of the Internal Execution Flow. Examples
+Provisioners are ceph-ansible, ceph-installer and gdeploy.
+
+Provisioning Module::
+This is a collection of various Provisioner Interface Modules and Integration
+Modules that tendrl ships with. This can be delivered as a dependency of the
+node agent package.
+
+Provisioner Interface Module::
+This is the ansible playbook that ships with tendrl core itself which enables
+tendrl core to know that it supports deployments for a specific storage system
+type and version, either for creation or import or both. The module would
+contain the necessary installation and configuration information to be able to
+deploy the Provisioner. It also contains the necessary YAML definitions that
+allow tendrl core to define, invoke and expose (via the api) the various
+provisioning operations, including creation and importing of clusters. In
+addition, the module also contains the code to enable the node agent to
+interface with the Provisioner.
+
+Integration Module::
+This is the ansible playbook that ships with tendrl core itself which enables
```

Comment:
This is currently taken care as part of ImportCluster flow. Do you mean this applicable only for create cluster flow? If so, that should be clearly mentioned here.